tests/integration_tests/test_collection_embedding_function.py

```
@@ -384,6 +390,454 @@ def test_get_or_create_collection_both_none_error(self, db_client):
+        """
+        Test that custom embedding functions are persisted and automatically restored.
+
+        Tests:
+        - Create collection with registered custom embedding function
+        - Get collection and verify embedding function is automatically restored
+        - Verify restored embedding function works correctly
+
+        Automatically runs for: embedded, server, oceanbase
+        """
+        collection_name = f"test_ef_persistence_{int(time.time() * 1000)}"
+        print(f"\n✅ Testing custom embedding function persistence and restoration")
```

Ruff (0.14.13)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)